common/src/test/java/com/github/dominik48n/party/server/SwitchServerTest.java

```diff
@@ -0,0 +1,214 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.dominik48n.party.server;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.dominik48n.party.api.Party;
+import com.github.dominik48n.party.api.PartyAPI;
+import com.github.dominik48n.party.api.PartyProvider;
+import com.github.dominik48n.party.api.player.PartyPlayer;
+import com.github.dominik48n.party.config.SwitchServerConfig;
+import com.github.dominik48n.party.user.UserManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.platform.commons.util.ReflectionUtils;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import static java.util.List.of;
+import static org.junit.platform.commons.util.ReflectionUtils.HierarchyTraversalMode.TOP_DOWN;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.never;
+
+@ExtendWith(MockitoExtension.class)
+class SwitchServerTest {
+
+    private SwitchServer<String> switchServer;
+
+    @Mock
+    private UserManager<String> userManager;
+
+    @Mock
+    private SwitchServerConfig switchServerConfig;
+
+    @Mock
+    private PartyPlayer partyPlayer;
+
+    @Mock
+    private PartyProvider partyProvider;
+
+    @BeforeEach
+    void setUp() throws IllegalAccessException {
+        setPartyProvider();
+        switchServer = createSwitchServer();
+    }
+
+    @Test
+    void handleServerConnected() throws JsonProcessingException {
+        String user = "USER";
+        String serverName = "Lobby3";
+        UUID leaderUUID = UUID.randomUUID();
+        UUID partyUuid = UUID.randomUUID();
+        Party party = createParty(partyUuid, leaderUUID);
+
+        when(userManager.getPlayer(user)).thenReturn(Optional.of(partyPlayer));
+        when(partyPlayer.partyId()).thenReturn(Optional.of(partyUuid));
+        when(partyPlayer.uniqueId()).thenReturn(leaderUUID);
+        when(partyProvider.getParty(partyUuid)).thenReturn(Optional.of(party));
+
+        when(switchServerConfig.blackEnable()).thenReturn(true);
+        when(switchServerConfig.blackPatternList()).thenReturn(of(Pattern.compile("^Lobby.*")));
+        when(switchServerConfig.whiteEnable()).thenReturn(true);
+        when(switchServerConfig.whitePatternList()).thenReturn(of(Pattern.compile("Lobby3")));
+
+        switchServer.handleServerConnected(user, serverName);
+
+        verify(partyProvider).connectPartyToServer(party, serverName);
+    }
+
+    @Test
+    void handleServerConnectedFailBecauseNotMatchWhiteList() throws JsonProcessingException {
```

Comment:
Please declare the variables in the method that should not be changed as `final`.